src/code-templates/libraries/logger/definition.ts

```diff
@@ -1,10 +1,10 @@
 export type LOG_LEVELS = 'debug' | 'info' | 'warn' | 'error' | 'critical';
 
 export interface Logger {
-  info(message: string, ...args: unknown[]): void;
-  error(message: string, ...args: unknown[]): void;
-  debug(message: string, ...args: unknown[]): void;
-  warning(message: string, ...args: unknown[]): void;
+  info(message: string, metadata?: Record<any, any>): void;
```

Comment:
Why not ```metadata? : {}```?

I'm not suggesting rather asking, not sure what exactly Pino is expecting for

If Record, why not <unknown, unknown>?

cc @mikicho

Reply:
metadata: Object would make sense here, but there are subtle differences between the two so need to check .. but I prefer metadata: Object

Reply:
Big O' Object means it must inherit from Object.proptotype which is not the case here

Record feels to me like an overkill, we want to educate for simplicity

```metadata: object``` feels simple, sufficient, and aligned with Pino's expectation

Reply:
I used Record as I remember having but here I don't get it so I'll use object :)
```
Don't use `object` as a type. The `object` type is currently hard to use ([see this issue](https://github.com/microsoft/TypeScript/issues/21732)).
Consider using `Record<string, unknown>` instead, as it allows you to more easily inspect and use the keys.eslint@typescript-eslint/ban-types
```

Reply:
[Not sure if the differences are relevant here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBBMF4YGUCeBbARiANgCgHIBDAgSgFgAoUSWAByNRxCIBNEYBvOALhgG0AjAF0AvlQBmAVzDAoAS3Ax5YCSABKAU1AAnVnnSaIEIgHNNfaDpWmANDENQ2RJwH4+W3awA8Vm-aIwVAA+Ui4YGghcTQA6ZlMDTSdWFyIw8UoVNU8QPUIAC01UAnsGJhZWCkpJGTlFMGVVEAB5TAArbShE4zMLOChrMDsHJOc3PlaOuTDuSOi4kATHMbSYDKyW9s6CopKYMuY2KprZBSUNkC25bpNzSwH-EeTU9xhLqagZiPAonFj4xLPJzpKgXK5dAiFYqlRiHSpAA). I think we should go with simplicity/readability. if issue will raise in the future, we'll tackle it than

Reply:
agree and updated